challenges/2022-10-11-blast-from-the-past/solutions/python/justinmemphis/justinmemphis.py

```diff
@@ -0,0 +1,67 @@
+"""
+# Find Three Largest Numbers
+Write a function that takes in an array of at least three integers and, without
+sorting the input array, returns a sorted array of the three largest integers in
+the input array. The function should return duplicate integers if necessary.
+## Business Rules/Errata
+ - The input array should have at least three integers. If it does not, you should
+   return a null value.
+ - You may not sort the input array
+ - The function should handle duplicate integers; for example `[10, 5, 9, 10, 12]`
+   should return `[10, 10, 12]`
+ - Constant space -> you will return a new array of 3 integers, and this will be the
+   only new data structure you create.
+ - Linear time -> you should solve this problem by only passing through the array a
+   single time.
+"""
+import unittest
+
+def Solution(array:list) -> list:
+    #print('function running')
+    bigList = [float('-inf'),float('-inf'),float('-inf')]
+    #print(type(bigList))
+
+    if len(array) < 3:
+        return None
+ 
+    for x in array:
+        # print(x)
+        if x > bigList[0]:
+            bigList[0] = x
+        if bigList[0] > bigList[1]:
+            temp = bigList[0]
+            bigList[0] = bigList[1]
+            bigList[1] = temp
+        if bigList[1] > bigList[2]:
+            temp = bigList[1]
+            bigList[1] = bigList[2]
+            bigList[2] = temp
+       # print(bigList)
+    
+    return bigList
+
+#Solution([1,2,3,4])
+#Solution([1])
+
+class TestSolution(unittest.TestCase):
+    """Test suite for Solution function"""
+
+    def setUp(self):
+        """ Convention """
+        self.test_normal_list = [141, 2, 17, -7, -17, -27, 18, 541, 8, 7, 7]
+        self.test_duplicates_in_soln = [11, -7, 5, -7]
+        self.test_none_ret = [1]
+    
+    def test_solution_normal_list(self):
+        self.assertListEqual(Solution(self.test_normal_list), [18, 141, 541])
+
+    def test_duplicates_in_soln(self):
+        self.assertListEqual(Solution([11, -7, 5, -7]), [-7, 5, 11])
+
+    def test_none_ret(self):
+        self.assertIsNone(Solution([1]))
```

Comment:
should reference self.test_none_ret or remove that from setup

Reply:
I prefer using `self.test_none_ret` because your test cases should be self explanatory. I should be able to read them like a document and tell what is going on. I think:
```
self.assertIsNone(Solution(self.test_none_ret))
```
is more intuitive to the reader than
```
self.assertIsNone(Solution([1]))
```
The attribute `self.test_none_ret` literally tells me what I should expect as a result.

Reply:
Thanks, I changed the reference.